test/functional/neutronless/bigip_interaction.py

```diff
@@ -197,26 +188,74 @@ def check_resulting_cfg(cls, test_name=current_test):
 
         test_name := the name of the test currently in tearDown
         """
-        cls.check_symbols()
-        cls._resulting_bigip_cfg(test_name)
+        if test_name:
+            cls.__current_test = test_name
+        else:
+            test_name = cls.__current_test
+        if not hasattr(pytest.symbols, 'no_bigip_tracking') and cls.ssh_cmd:
+            cls._resulting_bigip_cfg(test_name)
+        elif not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                not cls.ssh_cmd:
+            raise EnvironmentError("Broken state... cannot track without "
+                                   "ssh_cmd!")
 
     @classmethod
-    def backup_bigip_cfg(cls):
+    def backup_bigip_cfg(cls, test_name):
         """Performs a config backup of the BIG-IP's configuration
 
         This method will store a backup of the BIG-IP's configuration on the
         BIG-IP for later restoration.
         """
-        if not os.path.isfile(cls.config_file.format(my_epoch)):
-            cls.check_symbols()
+        cls.__current_test = test_name
+        if not os.path.isfile(cls.config_file.format(my_epoch)) and \
+                not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                cls.ssh_cmd:
             cls.__exec_shell(
                 cls.__ucs_cmd_fmt.format(cls.ssh_cmd, 'save'), True)
             cls._get_existing_bigip_cfg()
+        elif not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                not cls.ssh_cmd:
```

Comment:
Is this "elif" necessary, given the contents of the "BEGIN" function (ie. it seems like BEGIN is intended to prevent this condition)?